src/main/java/io/reactivex/internal/operators/single/SingleFlatMapPublisher.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2016-present, RxJava Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex.internal.operators.single;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import io.reactivex.Flowable;
+import io.reactivex.SingleObserver;
+import io.reactivex.SingleSource;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.exceptions.Exceptions;
+import io.reactivex.functions.Function;
+import io.reactivex.internal.functions.ObjectHelper;
+import io.reactivex.internal.subscriptions.SubscriptionHelper;
+
+public final class SingleFlatMapPublisher<S, T> extends Flowable<T> {
+
+    final SingleSource<S> source;
+    final Function<? super S, ? extends Publisher<? extends T>> mapper;
+
+    public SingleFlatMapPublisher(SingleSource<S> source,
+            Function<? super S, ? extends Publisher<? extends T>> mapper) {
+        this.source = source;
+        this.mapper = mapper;
+    }
+
+    @Override
+    protected void subscribeActual(Subscriber<? super T> actual) {
+        source.subscribe(new SingleFlatMapPublisherObserver<S, T>(actual, mapper));
+    }
+
+    static final class SingleFlatMapPublisherObserver<S, T> extends AtomicLong
+            implements SingleObserver<S>, Subscriber<T>, Subscription {
```

Comment:
Please `FlowableSubscriber`, otherwise the safety layer is added as a mapped `Flowable` will think it talks to an arbitrary consumer.

Reply:
Thanks, I'll go and check my other libraries for that one.